src/storage/engine/mod.rs

```diff
@@ -73,6 +73,9 @@ pub trait Engine: Send + Sync + Debug {
     fn delete_cf(&self, ctx: &Context, cf: CfName, key: Key) -> Result<()> {
         self.write(ctx, vec![Modify::Delete(cf, key)])
     }
+
+    /// Create a share Engine pointer.
+    fn clone(&self) -> Box<Engine + 'static>;
```

Comment:
Engine is not `Sized`, so here either return a trait object or generic. And return Self will face the same problem when using `Clone`.